File: ContainerWithMostWater.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        ans = 0
        left,right =0,len(height)-1

        while left < right:
            area = (right-left)* min(height[right],height[left])
            ans = max(ans,area)
            if height[right] <= height[left]:
                right -= 1
            else:
                left += 1

        return ans 

File: ContainerWithMostWater.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int ans = 0,area;
        int left=0; int right= height.size() -1;
        while (left < right){
            if(height[right] <= height[left]){
                area = (right-left)*height[right];
                right -= 1;
            }
            else{
                area = (right-left)*height[left];
                left += 1;
            }
            ans = (area > ans)? area : ans;

        }
        return ans;
    }
};
